interface/src/app/gene/genes.js

```diff
@@ -0,0 +1,376 @@
+angular.module('adage.gene.search', [
+  'adage.gene.resource'
+])
+
+.config(function($stateProvider) {
+  $stateProvider
+    .state( 'gene_search', {
+      url: '/gene_search',
+      views: {
+        "main": {
+          templateUrl: 'gene/gene-network.tpl.html',
+          controller: ['$scope', 'UserFactory',
+            function($scope, UserFactory) {
+              UserFactory.getPromise().$promise.then( function() {
+                $scope.userObj = UserFactory.getUser();
+              });
```

Comment:
@mhuyck : Aha, I see what you are saying. I thought this would be covered by the UserFactory setting the user to null by default (before calling the promise or anything else), but I see that the 'then' part of this code would not be executed, if, for example suddenly the server goes down (which would obviously break other things as well) or the tribe_client urls stopped working. However, the ng-if in the gene-network.tpl.html would resolve undefined (like null) to false, so it would also not show the geneset search form. I can, however, add on an error function (as @dongbohu was saying) to this to make sure the user is set to null if the promise totally breaks. Do you think this is a good approach?